update Libdc1394-dev for newer Ubuntu releases

FranekStark opened the pull request:
<!-- ROSDEP_RULE_TEMPLATE: Submitter Please review the contributing guidelines: https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md -->

Please add the following dependency to the rosdep database.

## Package name:

libdc1394-dev

## Purpose of using this:
Since Ubuntu kinetic the package `libdc1394-22-dev` is replaced by `libdc1394-dev`. 
Therefore I set `libdc1394-dev` as the new standard package for Ubuntu and added rules for the older Ubuntu releases.
See:
- https://askubuntu.com/questions/1407580/unable-to-locate-package-libdc1394-22-dev
- https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=963924


closes [#34921](https://github.com/ros/rosdistro/issues/34921)

## Links to Distribution Packages
- Ubuntu: https://packages.ubuntu.com/
   - https://packages.ubuntu.com/jammy/libdc1394-dev
   - https://packages.ubuntu.com/kinetic/libdc1394-dev

FranekStark committed:
updat libdc1394-dev for ubuntu versions newer than kinetic

FranekStark committed:
rosdep libdc1394-dev for ubuntu artful

FranekStark committed:
put libdc1394-dev packages in alphabetical order

nuclearsandwich reviewed (changes requested):
I suggest dropping the out-of-support distributions to keep things maintainable. It would also be nice to update the Debian keys equivalently (at least for bullseye and bookworm) but I can't use the suggestion API because those lines weren't touched.

FranekStark commented on rosdep/base.yaml:
```diff
@@ -3096,7 +3096,14 @@ libdc1394-dev:
+    'impish': [libdc1394-22-dev]
+    'focal': [libdc1394-22-dev]
+    'bionic': [libdc1394-22-dev]
+    'artful': [libdc1394-22-dev]
```
ok thanks for the comment.
Yes makes sense, I wasn't sure how outdated distributions are handled. So I tried to add as many old distributions I found to be backwards compatible. 
Also I will add a new commit for debian in a moment

FranekStark committed:
libdc1394-dev remove outdated distros

FranekStark committed:
libdc1394-dev change std package for debian distros

- https://packages.debian.org/stretch/libdc1394-22-dev
- https://packages.debian.org/buster/libdc1394-22-dev
- https://packages.debian.org/bullseye/libdc1394-dev
- https://packages.debian.org/bookworm/libdc1394-dev

FranekStark commented:
@nuclearsandwich I removed the outdated ubuntu keys, and added the debian keys.

clalancette commented on rosdep/base.yaml:
```diff
@@ -3088,15 +3088,21 @@ libdc1394-dev:
-  debian: [libdc1394-22-dev]
+  debian:
+    '*': [libdc1394-dev]
+    'buster': [libdc1394-22-dev]
+    'stretch': [libdc1394-22-dev]
```
While we aren't totally consistent about this, we don't typically quote these:

```suggestion
    buster: [libdc1394-22-dev]
    stretch: [libdc1394-22-dev]
```

clalancette commented on rosdep/base.yaml:
```diff
@@ -3088,15 +3088,21 @@ libdc1394-dev:
-  ubuntu: [libdc1394-22-dev]
+  ubuntu:
+    '*': [libdc1394-dev]
+    'focal': [libdc1394-22-dev]
+    'bionic': [libdc1394-22-dev]
```
Same here:

```suggestion
    focal: [libdc1394-22-dev]
    bionic: [libdc1394-22-dev]
```

clalancette reviewed (changes requested):
Two small things to change, then I'm happy to approve this.

FranekStark committed:
remove unececarry quotes in libdc1394-dev debian

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

FranekStark committed:
remove unececarry quotes in libdc1394-dev ubuntu

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

clalancette commented on rosdep/base.yaml:
```diff
@@ -3088,15 +3088,21 @@ libdc1394-dev:
-  ubuntu: [libdc1394-22-dev]
+  ubuntu:
+    '*': [libdc1394-dev]
+    focal: [libdc1394-22-dev]
+    bionic: [libdc1394-22-dev]
```
```suggestion
    bionic: [libdc1394-22-dev]
    focal: [libdc1394-22-dev]
```

clalancette reviewed (changes requested):
One more change should fix CI.

FranekStark committed:
reorder packag keys in debian libdc1394-dev

Co-authored-by: Chris Lalancette <clalancette@gmail.com>
